repository/transaction/transaction.go

```diff
@@ -1,41 +1,14 @@
 package transaction
 
 import (
-	"context"
 	"database/sql"
 	"errors"
-	"time"
-
-	"mock-payment-provider/primitive"
 )
 
 var ErrDuplicate = errors.New("duplicate")
 var ErrNotFound = errors.New("not found")
 var ErrExpired = errors.New("expired")
 
-type Transaction struct {
-	OrderId           string
-	TransactionAmount int64
-	PaymentType       primitive.PaymentType
-	TransactionStatus primitive.TransactionStatus
-	TransactionTime   time.Time
-}
-
-type ITransactionRepository interface {
-	// Migrate the database
-	Migrate(ctx context.Context) error
-	// Create creates a new entry of transaction. If OrderId already exists,
-	// it will return ErrDuplicate
-	Create(ctx context.Context, orderId string, amount int64, paymentType primitive.PaymentType, status primitive.TransactionStatus, expiredAt time.Time) error
```

Comment:
I think struct would be better for this function parameter. `Create(ctx context.Context, params CreateParams) error`. https://refactoring.guru/smells/long-parameter-list